src/ElasticsearchEngine.php

```diff
@@ -200,32 +179,29 @@ public function mapIds($results)
     {
         return collect($results['hits']['hits'])->pluck('_id')->values();
     }
-
-    /**
+   
+   /**
      * Map the given results to instances of the given model.
      *
+     * @param  \Laravel\Scout\Builder  $builder
      * @param  mixed  $results
      * @param  \Illuminate\Database\Eloquent\Model  $model
-     * @return Collection
+     * @return \Illuminate\Database\Eloquent\Collection
      */
-    public function map($results, $model)
+    public function map(Builder $builder, $results, $model)
```

Comment:
This change makes it compatible with the new abstract method signature of `map`, but the `$builder` is not passed on to `getScoutModelsByIds` to allow Scout to actually execute the query-callback that was added in Scout 5.

Reply:
Thanks for your comment.In last commit, `$builder` is passed on to `getScoutModelsByIds` to allow Scout to actually execute the query-callback that was added in Scout 5.